gossipd/gossip_generation.c

```diff
@@ -46,7 +47,8 @@ static u8 *create_node_announcement(const tal_t *ctx, struct daemon *daemon,
 	/* Add discovered IPs v4/v6 verified by peer `remote_addr` feature. */
 	/* Only do that if we don't have any addresses announced or
 	 * `config.ip_discovery` is explicitly enabled. */
-	if (count_announceable == 0 || daemon->ip_discovery) {
+	if ((daemon->ip_discovery == OPT_AUTOBOOL_AUTO && count_announceable == 0) ||
+	     daemon->ip_discovery == OPT_AUTOBOOL_TRUE) {
```

Comment:
Due to the enum ordering you can also say `daemon->ip_discovery >= OPT_AUTOBOOL_TRUE` :-)

Reply:
Hm, in this casenot really since the `count_announceable` also needs to be checked only for the `OPT_AUTOBOOL_AUTO` setting. Or do I get it wrong?

Reply:
Oh I see, you're right, I hadn't seen the nesting of those parens :-)